lib/streamlit/elements/text_widgets.py

```diff
@@ -84,8 +85,9 @@ def text_input(
         *,  # keyword-only arguments:
         placeholder: Optional[str] = None,
         disabled: bool = False,
+        clearable: bool = False,
         label_visibility: LabelVisibility = "visible",
-    ) -> str:
+    ) -> Union[str, None]:
```

Comment:
We probably need some overloads here. Otherwise, this will have a bit of a negative impact on the typing behavior. E.g. only if initial value is `None`, it returns an `Union[str, None]` otherwise it returns `str`.

Reply:
Hey, just tried to the the following overloads:

```
    @overload
    def text_input(
        self,
        label: str,
        value: SupportsStr,
        max_chars: Optional[int],
        key: Optional[Key],
        type: Literal["default", "password"],
        help: Optional[str],
        autocomplete: Optional[str],
        on_change: Optional[WidgetCallback],
        args: Optional[WidgetArgs],
        kwargs: Optional[WidgetKwargs],
        *,  # keyword-only arguments:
        placeholder: Optional[str],
        disabled: bool,
        clearable: bool,
        label_visibility: LabelVisibility,
    ) -> str:
        pass

    @overload
    def text_input(
        self,
        label: str,
        value: None,
        max_chars: Optional[int],
        key: Optional[Key],
        type: Literal["default", "password"],
        help: Optional[str],
        autocomplete: Optional[str],
        on_change: Optional[WidgetCallback],
        args: Optional[WidgetArgs],
        kwargs: Optional[WidgetKwargs],
        *,  # keyword-only arguments:
        placeholder: Optional[str],
        disabled: bool,
        clearable: bool,
        label_visibility: LabelVisibility,
    ) -> Union[str, None]:
        pass
```

I've tried multiple variations of the above, but `mypy` keeps failing with the error:
```
 No overload variant matches argument type "str" 
Possible overload variants:
e2e/scripts/app_hotkeys.py:17:1: note:     def text_input(self, label: str, value: SupportsStr, max_chars: Optional[int], key: Union[str, int, None], type: Literal['default', 'password'], help: Optional[str], autocomplete: Optional[str], on_change: Optional[Callable[..., None]], args: Optional[Tuple[Any, ...]], kwargs: Optional[Dict[str, Any]], *, placeholder: Optional[str], disabled: bool, clearable: bool, label_visibility: Literal['visible', 'hidden', 'collapsed']) -> str
e2e/scripts/app_hotkeys.py:17:1: note:     def text_input(self, label: str, value: None, max_chars: Optional[int], key: Union[str, int, None], type: Literal['default', 'password'], help: Optional[str], autocomplete: Optional[str], on_change: Optional[Callable[..., None]], args: Optional[Tuple[Any, ...]], kwargs: Optional[Dict[str, Any]], *, placeholder: Optional[str], disabled: bool, clearable: bool, label_visibility: Literal['visible', 'hidden', 'collapsed']) -> Optional[str]

```

I am not sure how to handle overloads in this case, seems like it is no enough difference for `mypy` to recognize this